src/org/zaproxy/zap/extension/callback/ui/CallbackTableModel.java

```diff
@@ -0,0 +1,189 @@
+ * limitations under the License.
+ */
+
+package org.zaproxy.zap.extension.callback.ui;
+
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.view.View;
+
+import javax.swing.table.AbstractTableModel;
+import java.awt.EventQueue;
+import java.util.ArrayList;
+import java.util.Date;
+
+public class CallbackTableModel extends AbstractTableModel {
+
+    private static final long serialVersionUID = 1L;
+    private final ArrayList<Column<CallbackModel>> columns;
+    private ArrayList<CallbackModel> callbackModels;
+
+    public CallbackTableModel() {
+        super();
+        columns = new ArrayList<>();
+        columns.add(createDateColumn());
+        columns.add(createMethodColumn());
+        columns.add(createUrlColumn());
+        columns.add(createHandlerColumn());
+        columns.add(createRefererColumn());
+        callbackModels = new ArrayList<>();
+    }
+
+    @Override
+    public int getColumnCount() {
+        return columns.size();
+    }
+
+    @Override
+    public String getColumnName(int column) {
+        return columns.get(column).getName();
+    }
+
+    @Override
+    public int getRowCount() {
+        return callbackModels.size();
+    }
+
+    @Override
+    public Object getValueAt(int row, int col) {
+        if (row >= callbackModels.size() || col >= columns.size()) {
+            return null;
+        }
+
+        CallbackModel model = getCallbackModelAt(row);
+        return columns.get(col).getValue(model);
+    }
+
+    @Override
+    public Class<?> getColumnClass(int columnIndex) {
+        if (columnIndex >= columns.size()) {
+            return null;
+        }
+
+        return columns.get(columnIndex).getColumnClass();
+    }
+
+    public CallbackModel getCallbackModelAt(int rowIndex){
+        return callbackModels.get(rowIndex);
+    }
+
+    public void addCallbackModel(final CallbackModel callbackModel) {
+        if (View.isInitialised() && !EventQueue.isDispatchThread()) {
+            EventQueue.invokeLater(new Runnable() {
+
+                @Override
+                public void run() {
+                    addCallbackModel(callbackModel);
+                }
+            });
+
+            return;
+        }
+        callbackModels.add(callbackModel);
+        int index = callbackModels.size() - 1;
+        this.fireTableRowsInserted(index, index);
+    }
+
+    public void removeAllCallbackModels() {
+        if (View.isInitialised() && !EventQueue.isDispatchThread()) {
+            EventQueue.invokeLater(new Runnable() {
+                @Override
+                public void run() {
+                    removeAllCallbackModels();
+                }
+            });
+            return;
+        }
+
+        int lastIndex = Math.max(callbackModels.size() - 1, 0);
+        callbackModels.clear();
+        callbackModels.trimToSize();
+        this.fireTableRowsDeleted(0, lastIndex);
```

Comment:
This will lead to an exception when empty (`IndexOutOfBoundsException: Invalid range`).